Directory.Build.targets

```diff
@@ -86,9 +86,10 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <!-- when building from source we need to use the current version of MetadataLoadContext as the toolset version, but source-build imports
-         another props file which overrides the SystemReflectionMetadataLoadContextVersion from Version.props so we can't set it there -->
+    <!-- when building from source we need to use the current version of various packages as the toolset version, but source-build imports
+         another props file which overrides the versions from Version.props so we can't set it there -->
     <SystemReflectionMetadataLoadContextToolsetVersion Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(SystemReflectionMetadataLoadContextVersion)</SystemReflectionMetadataLoadContextToolsetVersion>
+    <SystemTextJsonToolsetVersion Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(SystemTextJsonVersion)</SystemTextJsonToolsetVersion>
```

Comment:
should we override `SystemCollectionsImmutableToolsetVersion` and `SystemReflectionMetadataToolsetVersion` too?

Reply:
We have source-build-reference-packages for them for the versions we use, so it's not necessary. We could for consistency.

Reply:
MSBuild loads these ones in this case (that's why we're trying to version-match)

Reply:
I think I'd prefer the consistency